monai/metrics/panoptic_quality.py

```diff
@@ -0,0 +1,224 @@
+    https://openaccess.thecvf.com/content_CVPR_2019/papers/Kirillov_Panoptic_Segmentation_CVPR_2019_paper.pdf
+
+    Example of the typical execution steps of this metric class follows :py:class:`monai.metrics.metric.Cumulative`.
+
+    Args:
+        reduction: define mode of reduction to the metrics,
+            available reduction modes: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``},
+            default to ``"mean"``. if "none", will not do reduction.
+        remap: whether to remap `y_pred` and `y` to ensure contiguous ordering of instance id.
+        match_iou: IOU threshould to determine the paring between `y_pred` and `y`. Usually,
+            it should >= 0.5, the paring between instances of `y_pred` and `y` are identical.
+            If set `match_iou` < 0.5, this function uses Munkres assignment to find the
+            maximal amout of unique paring.
+        smooth_nr: a small constant added to the numerator to avoid zero.
+
+    """
+
+    def __init__(
+        self,
+        reduction: Union[MetricReduction, str] = MetricReduction.MEAN,
+        remap: bool = True,
+        match_iou: float = 0.5,
+        smooth_nr: float = 1e-6,
+    ) -> None:
+        super().__init__()
+        self.reduction = reduction
+        self.remap = remap
+        self.match_iou = match_iou
+        self.smooth_nr = smooth_nr
+
+    def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor):  # type: ignore
+        """
+        Args:
+            y_pred: input data to compute, typical panoptic segmentation model output.
+                It must be in the form of B1HW[D] and has integer type.
+            y: ground truth. It must has the same shape as `y_pred` and has integer type.
+
+        Raises:
+            ValueError: when `y_pred` and `y` have different shapes.
+            ValueError: when `y_pred` and `y` have > 1 channels.
+            ValueError: when `y_pred` and `y` have less than four dimensions.
+
+        """
+        if y_pred.shape != y.shape:
+            raise ValueError(f"y_pred and y should have same shapes, got {y_pred.shape} and {y.shape}.")
+
+        if y_pred.shape[1] != 1:
+            raise ValueError(
+                f"for panoptic quality calculation, only 1 channel input is supported, got {y_pred.shape[1]}."
+            )
+
+        dims = y_pred.ndimension()
+        if dims < 4:
+            raise ValueError(f"y_pred should have at least 4 dimensions (batch, channel, h, w), got {dims}.")
+
+        batch = y_pred.shape[0]
+        outputs = torch.zeros([batch, 1])
+        for b in range(batch):
+            outputs[b] = compute_panoptic_quality(
+                pred=y_pred[b, 0], gt=y[b, 0], remap=self.remap, match_iou=self.match_iou, smooth_nr=self.smooth_nr
+            )
+
+        return outputs
+
+    def aggregate(self, reduction: Union[MetricReduction, str, None] = None):
+        """
+        Execute reduction logic for the output of `compute_meandice`.
```

Comment:
Execute reduction logic for the output of `**compute_meandice**`. should be '**compute_panoptic_quality**'